Source/Scene/modelMaterialsCommon.js

```diff
@@ -0,0 +1,792 @@
+        var meshes = gltf.meshes;
+        for (var name in meshes) {
+            if (meshes.hasOwnProperty(name)) {
+                var mesh = meshes[name];
+                var primitives = mesh.primitives;
+                var primitivesCount = primitives.length;
+                for (var i=0;i<primitivesCount;++i) {
+                    var primitive = primitives[i];
+                    if (primitive.material === materialId) {
+                        var jointCount = 0;
+                        if (defined(primitive.attributes.JOINT)) {
+                            var nodes = gltf.nodes;
+                            for (var nodeName in nodes) {
+                                if (nodes.hasOwnProperty(nodeName)) {
+                                    var node = nodes[nodeName];
+                                    // We have skinning for this node and it contains this mesh
+                                    if (defined(node.skeletons) && defined(node.meshes) &&
+                                            node.meshes.indexOf(name) !== -1) {
+                                        jointCount = node.skeletons.length;
+                                    }
+                                }
+                            }
+                        }
+
+                        return {
+                            attributes : Object.keys(primitive.attributes),
+                            jointCount : jointCount
+                        };
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Modifies gltf in place.
+     *
+     * @private
+     */
+    var modelMaterialsCommon = function(gltf) {
+        if (!defined(gltf)) {
+            return undefined;
+        }
+
+        var hasExtension = false;
+        var extensionsUsed = gltf.extensionsUsed;
+        if (defined(extensionsUsed)) {
+            var extensionsUsedCount = extensionsUsed.length;
+            for(var i=0;i<extensionsUsedCount;++i) {
+                if (extensionsUsed[i] === 'KHR_materials_common') {
+                    hasExtension = true;
+                    extensionsUsed.splice(i, 1);
+                    break;
+                }
+            }
+        }
+
+        if (hasExtension) {
+            if (!defined(gltf.programs)) {
+                gltf.programs = {};
+            }
+            if (!defined(gltf.shaders)) {
+                gltf.shaders = {};
+            }
+            if (!defined(gltf.techniques)) {
+                gltf.techniques = {};
+            }
+
+            var lightParameters = generateLightParameters(gltf);
+
+            var techniques = {};
+            var materials = gltf.materials;
+            var meshes = gltf.meshes;
+            for (var name in materials) {
+                if (materials.hasOwnProperty(name)) {
+                    var material = materials[name];
+                    var primitiveInfo = getPrimitiveInfo(name, gltf);
```

Comment:
Why not call `getPrimitiveInfo` inside the `if` block below since it is only used there?